src/controller.rs

```diff
@@ -40,6 +40,11 @@ impl Controller {
         // Start the dummy adapter.
         let dummy_adapter = DummyAdapter::new(self.sender.clone(), self.context.clone());
         dummy_adapter.start();
+
+        {
+            let mut context = self.context.lock().unwrap();
+            context.start_tunnel().unwrap();
+        }
```

Comment:
It's to keep the lock contained.  It's going out of scope, but I liked to keep it obvious in case something is added after and the scope of the lock is neglected.